diff --git a/dist/compiler/compileBrowser.js b/dist/compiler/compileBrowser.js
index 304f1b7444af5398a08a0fa8cd165cb4fb0f64eb..fc50cddb58e30fe11c7e5800fa41c25c79c6514b 100644
--- a/dist/compiler/compileBrowser.js
+++ b/dist/compiler/compileBrowser.js
@@ -130,6 +130,14 @@ const createEsbuildConfig = (build, config, options, onLoader) => {
       remixConfig: config
     }));
   }
+  plugins.push({
+    name: "nodejs_compat Plugin",
+    setup(pluginBuild) {
+      pluginBuild.onResolve({ filter: /node:.*/ }, () => {
+        return { external: true };
+      });
+    },
+  })
   return {
     entryPoints,
     outdir: config.assetsBuildDirectory,